### ISSUE TITLE
MONDOO_API_TOKEN, MONDOO_CONFIG_PATH or MONDOO_CONFIG_BASE64 need to be set

### ISSUE THREAD

**Issue by GurdipS5** [mondoohq/terraform-provider-mondoo#78]
**What is not working as you expected it?**
I am trying to run Terraform for the Mondoo provider on Windows 11. I have setup cnspec and there is a config file in C:\programdata\Mondoo. I created the relevant tokens in the portal. My provider looks like below:

`terraform {
  required_providers {
    mondoo = {
      source = "mondoohq/mondoo"
    }
  }
}

provider "mondoo" {
  region = "eu" # use "eu" for the European region
  credentials = "C:\\programdata\\Mondoo\\mondoo.yml"
}`

The error is:

Planning failed. Terraform encountered an error while generating this plan.

╷
│ Error: MONDOO_API_TOKEN, MONDOO_CONFIG_PATH or MONDOO_CONFIG_BASE64 need to be set
│ 
│   with provider["registry.terraform.io/mondoohq/mondoo"],
│   on provider.tf line 9, in provider "mondoo":
│    9: provider "mondoo" {
│
│ MONDOO_API_TOKEN, MONDOO_CONFIG_PATH or MONDOO_CONFIG_BASE64 need to be set

**Where on the platform does it happen?**
Local

**How do we replicate the issue?**
Replicate terraform

**Expected behavior (i.e. solution)**
Successful plan.

**Other Comments**

**Comment by chris-rock** [comment]
@GurdipS5 Thank you for the report. At this point we do not support setting the service account via the provider config. You just need to set the variable with the credential path then:

```
export MONDOO_CONFIG_PATH=/path/to/your/serviceaccount/youthful-meitner-435985.json
```

on Windows you would set it via:

```
$Env:MONDOO_CONFIG_PATH = "C:\programdata\Mondoo\mondoo.yml"
```

I'll mark your issue as an enhancement since I like the way to configure the credential path via terraform.

**Comment by preslavgerchev** [comment]
Hi @GurdipS5, thank you for reporting this.
This has been fixed with the latest (0.14.0) release. Do note the `credentials` is no longer the path, but the JSON service account (the contents of the file) directly. I have opened #124 as a follow-up to also support injecting the mondoo creds via a path.
